File: Dynamic_Programming/Climbing Stairs.cpp
class Solution {
public:
    
    
    int fibbo(int n,vector<int> &dp)
    {
        if(n<=2)
        {
            return n;
        }
	   if(dp[n] != -1) {
		return dp[n];
       }
	return dp[n] = fibbo(n-1,dp) + fibbo(n-2,dp);
    }
    
    
    
    
    
    
    
    
    
    int climbStairs(int n) {
        vector<int>dp(n+1,-1);
        
        return fibbo(n,dp);
        
         
        
        
    }
};

File: Dynamic_Programming/Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        a = 1
        b = 1
        c = a+b
        if n == 1:
            return 1
        elif n == 2:
            return 2
        else:
            for i in range(3,n+1):
                a = b
                b = c
                c = a+b
            return c
        
